modules/core/react/lib/InputProvider.tsx

```diff
@@ -150,16 +146,9 @@ export default class InputProvider extends React.Component<InputProviderProps, I
       console.warn('Browser does not support passive event listeners');
     }
 
-    this.state = {
-      currentInput: storedInput || InputType.Initial,
-      currentIntent: storedIntent || InputType.Initial,
-      supportsPassive: supportsPassive || false,
-      isBuffering: false,
-      isScrolling: false,
-      mousePosX: null,
-      mousePosY: null,
-      nested: false,
-    };
+    this.currentInput = storedInput || InputType.Initial;
```

Comment:
What was the reason for changing things from state to class variables? Was it to avoid the need for `shouldComponentUpdate`?

Reply:
i think the other reason is we didn't want to force a rerender every time state changed, so having class variables made more sense.